pkg/api/validation/validation_test.go

```diff
@@ -2881,6 +2924,121 @@ func TestAlphaHugePagesIsolation(t *testing.T) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "",
+		},
+		Spec: api.PersistentVolumeSpec{
+			Capacity: api.ResourceList{
+				api.ResourceName(api.ResourceStorage): resource.MustParse("10G"),
+			},
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadWriteOnce},
+			PersistentVolumeSource: api.PersistentVolumeSource{
+				HostPath: &api.HostPathVolumeSource{
+					Path: "/foo",
+					Type: newHostPathType(string(api.HostPathDirectory)),
+				},
+			},
+			StorageClassName: "test-storage-class",
+			VolumeMode:       &validMode,
+		},
+	}
+	invalidModePV := api.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "",
+		},
+		Spec: api.PersistentVolumeSpec{
+			Capacity: api.ResourceList{
+				api.ResourceName(api.ResourceStorage): resource.MustParse("10G"),
+			},
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadWriteOnce},
+			PersistentVolumeSource: api.PersistentVolumeSource{
+				HostPath: &api.HostPathVolumeSource{
+					Path: "/foo",
+					Type: newHostPathType(string(api.HostPathDirectory)),
+				},
+			},
+			StorageClassName: "test-storage-class",
+			VolumeMode:       &invalidMode,
+		},
+	}
+
+	// PersistentVolumeClaim with VolumeMode set (valid and invalid)
+	validModePVC := api.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "default",
+		},
+		Spec: api.PersistentVolumeClaimSpec{
+			Resources: api.ResourceRequirements{
+				Requests: api.ResourceList{
+					api.ResourceName(api.ResourceStorage): resource.MustParse("10G"),
+				},
+			},
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadWriteOnce},
+			StorageClassName: &validName,
+			VolumeMode:       &validMode,
+		},
+	}
+	invalidModePVC := api.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "default",
+		},
+		Spec: api.PersistentVolumeClaimSpec{
+			Resources: api.ResourceRequirements{
+				Requests: api.ResourceList{
+					api.ResourceName(api.ResourceStorage): resource.MustParse("10G"),
+				},
+			},
+			AccessModes: []api.PersistentVolumeAccessMode{api.ReadWriteOnce},
+			StorageClassName: &validName,
+			VolumeMode:       &invalidMode,
+		},
+	}
+
+	// Enable alpha feature BlockVolumeSupport
+	err := utilfeature.DefaultFeatureGate.Set("BlockVolumeSupport=true")
+	if err != nil {
+		t.Errorf("Failed to enable feature gate for BlockVolumeSupport: %v", err)
+		return
+	}
+	if errs := ValidatePersistentVolume(&validModePV); len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+	if errs := ValidatePersistentVolumeClaim(&validModePVC); len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+	if errs := ValidatePersistentVolume(&invalidModePV); len(errs) == 0 {
+		t.Errorf("expected failure: %v", errs)
+	}
+	if errs := ValidatePersistentVolumeClaim(&invalidModePVC); len(errs) == 0 {
+		t.Errorf("expected failure: %v", errs)
+	}
+
+
+	// Disable alpha feature BlockVolumeSupport
```

Comment:
These test cases are already covered by the first 2 you added in this file.  You can decide if you want to keep all the block-related tests in one function, or have them split.

Reply:
@msau42 - not sure I'm following, which "first 2 you added in this file" are you referring to?

Reply:
@msau42 - oh, you mean the ```"alpha disabled valid volume mode"``` in the pv and pvc tests...

Reply:
I was going back and forth on this, I ultimately thought it would be better to keep them separate since volumeDevice should be it's own similar to volumeMount and volumeMode really is part of pv and pvc validation